app/server/README.md

```diff
@@ -64,45 +62,250 @@ module.exports = {
+          "id": "button"
+        }
+      },
+      "args": { "color": "red", "label": "Stop" }
+    },
+    {
+      "name": "Green",
+      "parameters": {
+        "server": { 
+          "id": "button"
+        }
+      },
+      "args": { "color": "green", "label": "Go" }
+    }
+  ]
+}
+```
 
 ## Addon compatibility
 
 Storybook also comes with a lot of [addons](https://storybook.js.org/addons/introduction) and a great API to customize as you wish. As some addons assume the story is rendered in JS, they may not work with `@storybook/server` (yet!).
 
 Many addons that act on the manager side (such as `backgrounds` and `viewport`) will work out of the box with `@storybook/server` -- you can configure them with parameters written on the server as usual.
 
-Some addons work in specific ways:
+### Controls
 
-### Knobs
+To configure controls, simple add `args` and `argTypes` keys to the story JSON much like you would CSF:
 
-> Note this functionality will likely be deprecated in favor of args support in the near future.
-
-To configure knobs, add a special `"knobs"` key to the story JSON:
-
-```json:
-"knobs": [
-  { "type": "text", "name": "Name", "value": "John Doe", "param": "name"},
-  { "type": "number", "name": "Age", "value": 44, "param": "age"}
-]
+```json
+{
+  "title": "Controls",
+  "stories": [
+    {
+      "name": "Button",
+      "parameters": {
+        "server": { "id": "controls/button" }
+      },
+      "args": { "button_text": "Push Me", "color": "red" },
+      "argsTypes": { "button_text": { "control": { "type": "color" } } }
+    },
+  ]
+}
 ```
 
-The knobs values will be added to your story URL as query parameters.
+The controls values will be added to your story URL as query parameters.
 
 ### Actions
 
 To use actions, use the `parameters.actions.handles` parameter:
 
 ```json
-"actions": {
-  "handles": ["mouseover", "click .btn"]
+{
+  "title": "Actions",
+  "stories": [
+    {
+      "name": "Button",
+      "parameters": {
+        "server": { "id": "actions/button" },
+        "actions": {
+          "handles": ["mouseover", "click .btn"]
+        }
+      }
+    }
+  ]
 }
 ```
+
+## Advanced Configuraiton
+
+### fetchStoryHtml
+
+For control over how `@storybook/server` fetches Html from the server you can provide a `fetchStoryHtml` function as a parameter. You would typically set this in `.storybook/preview.js` but it's just a regular Storybook parameter so could be overriden at the stories or story level.
+
+
+```javascript
+// .storybook/preview.js
+
+const fetchStoryHtml = async (url, path, params) => {
+  // Custom fetch impelentation
+  // ....
+  return html;
+};
+
+export const parameters = {
+  server: {
+    url: `http://localhost:${port}/storybook_preview`,
+    fetchStoryHtml
+  },
+};
+```
+
+`fetchStoryHtml` should be an async function with the following signature
+
+```javascript
+type FetchStoryHtmlType = (url: string, id: string, params: any) => Promise<string | Node>;
+```
```

Comment:
I'm debating this API a little. At the moment server is doing the merging of parameters, args etc. as well as some simple control value -> string serialization logic. However, if the consumer wanted to change some/all of that they'd be stuck. The alternative is to make this `fetchStoryHtml` take args, argTypes and parameters. With that pattern the overrides can do anything they want but gain nope of the benefits of the existing implementation?

Thoughts on what we should push as v1? This API is a bit speculative, I expect the main usage of overriding to be adding things like authentication but I can equally well see frameworks needing particular formats of the date or a different url layout etc.